### PR TITLE
IGNITE-15979: Calcite engine. Script test runner ignores "statement error" queries result

### PR DESCRIPTION
**Pull request by lowka**

### CODE REVIEW

**Review by AMashenkov — APPROVED** [review]

File: modules/runner/src/integrationTest/java/org/apache/ignite/internal/sqllogic/SqlScriptRunner.java [hunk 1]
```diff
@@ -244,78 +289,58 @@ boolean ready() throws IOException {
...
+            return new ScriptPosition(fileName, lineNum);
         }
 
         @Override
         public void close() throws Exception {
             buffReader.close();
         }
 
+        @Nullable
         private Command nextCommand() {
```
Comment (AMashenkov) [inline modules/runner/src/integrationTest/java/org/apache/ignite/internal/sqllogic/SqlScriptRunner.java:302]:
```suggestion
        private @Nullable Command nextCommand() {
```

File: modules/runner/src/integrationTest/java/org/apache/ignite/internal/sqllogic/SqlScriptRunner.java [hunk 2]
```diff
@@ -244,78 +289,58 @@ boolean ready() throws IOException {
...
         }
 
         @NotNull
         @Override
         public Iterator<Command> iterator() {
-            final Command cmd0 = nextCommand();
+            Command cmd0 = nextCommand();
             return new Iterator<>() {
+                @Nullable
                 private Command cmd = cmd0;
```
Comment (AMashenkov) [inline modules/runner/src/integrationTest/java/org/apache/ignite/internal/sqllogic/SqlScriptRunner.java:344]:
```suggestion
                private @Nullable Command cmd = cmd0;
```

**Comment by AMashenkov** [comment]
Looks good.
Les't fix styles and extract static Command classes from SqlScriptRunner to the top-level classes.

File: modules/runner/src/integrationTest/java/org/apache/ignite/internal/sqllogic/SqlScriptRunner.java (hunk 1 above)
Comment (lowka) [reply modules/runner/src/integrationTest/java/org/apache/ignite/internal/sqllogic/SqlScriptRunner.java]: > Les't fix styles and extract static Command classes from SqlScriptRunner to the top-level classes.

Done + added javadocs and some test cases to _runner/self.test

File: modules/runner/src/integrationTest/java/org/apache/ignite/internal/sqllogic/SqlScriptRunner.java (hunk 2 above)
Comment (lowka) [reply modules/runner/src/integrationTest/java/org/apache/ignite/internal/sqllogic/SqlScriptRunner.java]: Fixed

File: modules/runner/src/integrationTest/java/org/apache/ignite/internal/sqllogic/SqlScriptRunner.java (hunk 1 above)
Comment (lowka) [reply modules/runner/src/integrationTest/java/org/apache/ignite/internal/sqllogic/SqlScriptRunner.java]: Fixed

**Comment by lowka** [comment]
> Looks good.
> Les't fix styles and extract static Command classes from SqlScriptRunner to the top-level classes.

Done + added javadocs and some test cases to _runner/self.test

File: modules/runner/src/integrationTest/java/org/apache/ignite/internal/sqllogic/SqlScriptRunner.java (hunk 1 above)
Comment (lowka) [reply modules/runner/src/integrationTest/java/org/apache/ignite/internal/sqllogic/SqlScriptRunner.java]: fixed

**Review by AMashenkov — APPROVED** [review]

**Pull request merged**